src/collections/componentSet.ts

```diff
@@ -296,6 +298,21 @@ export class ComponentSet extends LazyCollection<MetadataComponent> {
     return result;
   }
 
+  /**
+   * Gets the metafile path for a source component. Not all the components have an XML metafile,
+   * e.g., DigitalExperience has a JSON metafile (_meta.json).
+   *
+   * @param component The source component
+   * @returns The metafile path
+   */
+  public static getComponentMetaFilePath(component: SourceComponent): string {
+    if (['digitalexperience'].includes(component.type.id)) {
+      return join(dirname(component.content), META_JSON_FILE);
+    }
+
+    return component.xml;
+  }
+
```

Comment:
"_meta.json" is not a suffix, it's the name of the metafile and we need its path.

Reply:
FYI @Kaaviyan

Reply:
It's more flexible to add this property to the metadata registry as a suffix to be consumed in the code.  We can consider this a suffix with a null file name.  I don't expect there will be more types that change the well established pattern of `-meta.xml`, but if it happens this can used to build a file name of (e.g.) `<MyTypeName>_meta.json`.